core/metacling/src/TClingClassInfo.cxx

```diff
@@ -835,9 +822,16 @@ bool TClingClassInfo::IsValidMethod(const char *method, const char *proto,
 
 int TClingClassInfo::InternalNext()
 {
-
    R__LOCKGUARD(gInterpreterMutex);
 
+   cling::Interpreter::PushTransactionRAII RAII(fInterp);
+   // fDecl must be a DeclContext in order to iterate.
+   const clang::DeclContext *DC = cast<DeclContext>(fDecl);
+   if (fIterAll)
```

Comment:
Isn't there a missing conditional?  This looks like it restarts the iteration at each call to InternalNext.  What am I missing?

Reply:
Doh, true. We need to wrap it in `if (fFirstTime)`